jspoon/src/main/java/pl/droidsonroids/jspoon/ArrayUtils.java

```diff
@@ -0,0 +1,168 @@
+package pl.droidsonroids.jspoon;
+
+class ArrayUtils {
+
+    /**
+     * Converts a given array of wrappers to the same type of primitive array. Null values are
+     * represented as initial value for that primitive type. If the wrapper array is null, will
+     * return null.
+     * @param array wrapper array
+     * @param primitiveType type of the returned primitive array
+     * @throws IllegalArgumentException if the given primitive type is not a class of a primitive
+     * @throws ClassCastException if primitive type does not represent items of the wrapper array
+     */
+    static Object toPrimitive(final Object[] array, Class<?> primitiveType) {
+        if (array == null) {
+            return null;
+        }
+        if (!primitiveType.isPrimitive()) {
+            throw new IllegalArgumentException(primitiveType + " is not a primitive type");
+        }
+        Class<?> pt = primitiveType;
```

Comment:
If you're referring to `Class<?> pt = primitiveType;`  then No. Just a full name as argument for readablity and shorter for compact implementation.

If you're referring to `primitiveType` argument validation, then Yes. Because
```
Integer.TYPE.equals(Integer.class)); // false
Integer.TYPE.equals(int.class); // true
```

Reply:
I would say that `primitiveType` is more readable than `pt`.